### ISSUE TITLE
team_broadcast_ tests failing with HPX, Intel OpenMP nightly builds

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#3526]
Following merge of #3488 the `team_broadcast_{float,double}` unit tests began failing 10/24/2020 with Intel builds and OpenMP, Pthreads builds

The `team_broadcast_{long,char,float,double}` unit tests are failing in the nightly HPX builds with gcc/6.1

The `team_broadcast` test (prior to renaming in #3488 was passing with SHA 985e9d57fb35412b312e612717bd50a464efdc77 of the prior merge in commit history #3514

Sample error message (intel/19 openmp build):
```bash

[ RUN      ] openmp.team_broadcast_float
/ascldap/users/ndellin/kokkos/core/unit_test/TestTeam.hpp:1345: Failure
Value of: (double)B
  Actual: 6039002.5
Expected: (double)A
Which is: 6039033
/ascldap/users/ndellin/kokkos/core/unit_test/TestTeam.hpp:1345: Failure
Value of: (double)B
  Actual: 12078005
Expected: (double)A
Which is: 12078066
/ascldap/users/ndellin/kokkos/core/unit_test/TestTeam.hpp:1345: Failure
Value of: (double)B
  Actual: 6039002.5
Expected: (double)A
Which is: 6039033
/ascldap/users/ndellin/kokkos/core/unit_test/TestTeam.hpp:1345: Failure
Value of: (double)B
  Actual: 12078005
Expected: (double)A
Which is: 12078066
[  FAILED  ] openmp.team_broadcast_float (464 ms)
[ RUN      ] openmp.team_broadcast_double
/ascldap/users/ndellin/kokkos/core/unit_test/TestTeam.hpp:1343: Failure
Value of: (double)B
  Actual: 6039040.000000108
Expected: (double)A
Which is: 6039039.9999999916
/ascldap/users/ndellin/kokkos/core/unit_test/TestTeam.hpp:1343: Failure
Value of: (double)B
  Actual: 12078080.000000216
Expected: (double)A
Which is: 12078079.999999983
/ascldap/users/ndellin/kokkos/core/unit_test/TestTeam.hpp:1343: Failure
Value of: (double)B
  Actual: 6039040.000000108
Expected: (double)A
Which is: 6039039.9999999916
/ascldap/users/ndellin/kokkos/core/unit_test/TestTeam.hpp:1343: Failure
Value of: (double)B
  Actual: 12078080.000000216
Expected: (double)A
Which is: 12078079.999999983
[  FAILED  ] openmp.team_broadcast_double (457 ms)

```

Sample reproducer (blake SKX testbed):

```bash
module load intel/compilers/19.1.144 cmake/3.12.3

$KOKKOS_PATH/generate_makefile.bash     --arch=SKX     --cxxflags="-Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized"     --compiler=icpc     --with-devices=OpenMP --cxxstandard="14" --kokkos-path=$KOKKOS_PATH --no-examples

# Alternative cmake line
cmake -DCMAKE_CXX_COMPILER=`which icpc` -DCMAKE_CXX_FLAGS="-Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized" -DKokkos_ENABLE_OPENMP=ON -DKokkos_ARCH_SKX=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_BUILD_TYPE=RELEASE $KOKKOS_PATH

make -j

export OMP_PROC_BIND=spread
export OMP_PLACES=threads

cd core/unit_test
./KokkosCore_UnitTest_OpenMP --gtest_filter=openmp.team_broadcast*
```

**Comment by masterleinad** [comment]
Previously, we were only running the `team_broadcast` test with `T=long` disabling the other cases would likely be the simplest workaround.

**Comment by crtrott** [comment]
Looks like rounding issues?

**Comment by ndellingwood** [comment]
Addressed by #3573
